multiqc/modules/hops/hops.py

```diff
@@ -0,0 +1,131 @@
+""" MultiQC module to parse output from HOPS postprocessing script """
+
+from __future__ import print_function
+from collections import OrderedDict
+import logging
+import json
+
+from multiqc.plots import heatmap
+from multiqc.utils import config
+from multiqc.modules.base_module import BaseMultiqcModule
+
+log = logging.getLogger(__name__)
+
+class MultiqcModule(BaseMultiqcModule):
+    def __init__(self):
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(name='HOPS', anchor='hops',
+        href="https://www.https://github.com/rhuebler/HOPS/",
+        info="is an ancient DNA characteristics screening tool of output from the metagenomic aligner MALT.")
+
+        # Find and load any HOPS post-processing JSONs
+        self.hops_data = dict()
+
+        for f in self.find_log_files('hops',filehandles=True):
+            try:
+                self.parseJSON(f)
+            except KeyError:
+                logging.warning("Error loading file {}".format(f['fn']))
+
+        self.hops_data = self.ignore_samples(self.hops_data)
+
+        if len(self.hops_data) == 0:
+            raise UserWarning
+
+        log.info("Found {} samples".format(len(self.hops_data)))
+
+        # This type of data isn't 'summarise-able' for general stats, so 
+        # skipping straight to heatmap. We also won't write data file to the
+        # multiqc_data directory because it would be exactly same as input JSON. 
+        self.hops_heatmap()
+
+    def parseJSON(self, f):
+        """ Parse the JSON output from HOPS and save the summary statistics """
+
+        try:
+            parsed_json = json.load(f['f'])
+        except JSONDecodeError as e:
+            log.debug("Could not parse HOPS JSON: '{}'".format(f['fn']))
+            log.debug(e)
+            return None
+        
+        # Convert JSON to dict for easier manipulation
+        for s in parsed_json: 
+            s_name = self.clean_s_name(s, f['root'])
+            if s_name in self.hops_data: 
+                log.debug("Duplicate sample name found! Overwriting: {}".format(s_name))
+            self.add_data_source(f, s_name=s_name)
+            self.hops_data[s_name] = {}
+            for t in parsed_json[s]:
+                self.hops_data[s_name][t] = parsed_json[s][t][0]
```

Comment:
Regarding the `[0]` - I noticed that the test data had values in an array. Why is this? Can the files support more than one sample or something? Will that array ever have more than one value?

Reply:
No it shouldn't. That was a leftover from the JSON parsing, where each value was in it's own list (for some reason). Will remove.
```suggestion
                self.hops_data[s_name][t] = parsed_json[s][t]
```

Reply:
Yeah, it's the reason that I was curious about really 😉 Just thinking ahead if there could be some other user turning up with a JSON file that has multiple values in that list.

But are you guys writing the tool as well? So I guess you have control over that?

Reply:
Yup, well at least I am peripherally involved - I added the JSON support. It's extremely unlikely that there could be any more then one value - the JSON is simply covered from a R matrix (rows are taxa, Columns are samples)